clang/include/clang/CIR/Dialect/Builder/CIRBaseBuilder.h

```diff
@@ -442,6 +442,22 @@ class CIRBaseBuilderTy : public mlir::OpBuilder {
     return createCast(cir::CastKind::ptr_to_bool, v, getBoolTy());
   }
 
+  // TODO(cir): the following function was introduced to keep in sync with LLVM
+  // codegen. CIR does not have "zext" operations. It should eventually be
+  // renamed or removed. For now, we just add whatever cast is required here.
```

Comment:
I don't think we should follow this strategy upstream, particularly because it leaves the `llvm_unreachable` call below. I think for the purposes of this PR, you can just call `createBoolToInt` directly.